src/citrine/informatics/design_spaces/hierarchical_design_space.py

```diff
@@ -25,26 +25,30 @@ class TemplateLink(Serializable["TemplateLink"]):
         Citrine ID referencing an on-platform material template.
     process_template: UUID
         Citrine ID referencing an on-platform process template.
-    template_name: Optional[str]
-        Optional name describing the combination of templates for display purposes.
-        The Citrine Platform by default uses the name of the material template.
+    material_template_name: Optional[str]
+        Optional name of the material template.
+    process_template_name: Optional[str]
+        Optional name of the process template.
 
     """
 
     material_template = properties.UUID("material_template")
     process_template = properties.UUID("process_template")
-    template_name = properties.Optional(properties.String, "name")
+    material_template_name = properties.Optional(properties.String, "material_template_name")
+    process_template_name = properties.Optional(properties.String, "process_template_name")
 
     def __init__(
             self,
             *,
             material_template: UUID,
             process_template: UUID,
-            template_name: Optional[str] = None
```

Comment:
@jspeerless do you care that this is a breaking change? Nobody is really using this yet

Reply:
I'm fine with either option (minor bias toward deprecation cycle), but it'd be pretty straight forward to support both keyword arguments, issuing deprecation warnings on `template_name` usage and feeding them both into the material_template_name field.

Reply:
ill support both with an actual deprecation.